Update build to use Gradle7.5.1

nicost opened the pull request:
Tested on my Windows 10 box only.  Builds successfully (including tests).  The resulting ClearVolume.jar is functional (in Micro-manager on my Windows machine).  I removed the JavaFX part (no idea what it is for, and if anyone still uses it).  Also added the maven-publish plugin and configured it, but can not test that part as I do not have credentials.  

Please note that I am a complete novice at gradle, and this is a complete hack, just to make things work.  We need these builds to resume, so please use this as a basis to get things going again (hopefully easy based on this PR).

Commit:
Enables build with gradle 7.5.1 in Windows 64.
Complete hack, no idea what I am doing.
Capsule plugin is horribly outdated.  Replaces with shadow plugin that seems
a lot simpler.  No idea if it is doing the same thing.
javafx-gradle-plugin is unmaintained and completely outdated.  I simply
disabled all the java fx stuff.  No idea if anyone needs it or uses it.
Tests do pass, and the resulting binary works in Micro-Manager, so I suspect
it will also work in Fiji.  Next step would be to figure out the
maven-publish plugin, which should not be that hard.

Commit:
Added maven publishing section.  Seems to work, but I do not have the credentials to actually test.

skalarproduktraum commented:
Hey @nicost! Thanks for this valuable contribution, and apologies for not being in touch earlier :-/ The PR looks good, there's a few notes I have that I'll leave as review. With regard to Github CI, I'm happy to help out with this, we can probably take over most of the CI config we're using for scenery for that purpose.

skalarproduktraum commented on build.gradle:
```diff
@@ -14,17 +14,18 @@ buildscript {
   id "com.jfrog.bintray" version "1.2"
-  id "us.kirchmeier.capsule" version "0.9.0"
+  //id "us.kirchmeier.capsule" version "1.0.2"
+//  id 'com.github.johnrengelman.shadow' version '7.1.2'
```
Do we actually still need the Fat JAR? If not, all the capsule and shadow stuff can go.

skalarproduktraum commented on build.gradle:
```diff
@@ -14,17 +14,18 @@ buildscript {
 apply plugin: 'cpp'
 apply plugin: 'idea'
-apply plugin: 'javafx-gradle-plugin'
+//apply plugin: 'javafx-gradle-plugin'
```
This can also go away, it was just used for packaging in ye olden days.

skalarproduktraum replied:
Actually, only for packaging for Java Web Start, so not relevant anymore.

nicost replied:
Gone now.

skalarproduktraum commented on build.gradle:
```diff
@@ -90,44 +91,44 @@ configurations.all
+  implementation 'net.sf.trove4j:trove4j:3.0.3'
+  implementation 'args4j:args4j:2.0.29'
+  implementation 'com.apple:AppleJavaExtensions:1.4'
+  implementation 'com.miglayout:miglayout:3.7.4'
```
Miglayout should probably be upgraded to the latest version used in the scijava POM

skalarproduktraum replied:
Same for the commons stuff 👍

nicost replied:
I updated all dependencies to the same version as those in the scijiava POM (if present).  Miglayout has been split in 3.  Remarkably, no compiler errors.

skalarproduktraum commented on build.gradle:
```diff
@@ -90,44 +91,44 @@ configurations.all
 	
   //ClearCL:
-  compile ('net.clearvolume:clearcl:0.5.2') {transitive= true; changing= true}
+  implementation ('net.clearvolume:clearcl:0.5.2') {transitive= true; changing= true}
```
ClearCL and ClearAudio could potentially be made into `api` dependencies, instead of `implementation`. `implementation` means they're kinda private, while `api` will expose them.

skalarproduktraum replied:
Also, both `transitive` and `changing` options should be removed. `transitive` is true by default, and changing is only relevant for SNAPSHOTs.

nicost replied:
I changed them into api dependencies (which prompted addition of `apply plugin java-library`) and removed the `transitive `and `changing `options.

skalarproduktraum commented on build.gradle:
```diff
@@ -519,12 +505,30 @@ artifacts
 	}
 }
 
+// I guess this can be deleted as bintray is no longer
```
You have guessed correctly, just get rid of the Bintray stuff 👍

skalarproduktraum commented on build.gradle:
```diff
@@ -599,6 +605,7 @@ uploadArchives {
 
 // configure javafx-gradle-plugin
```
This can also go away.

nicost replied:
Done.

skalarproduktraum commented:
I also invited you to the organisation, so stuff like this becomes easier in the future 👍

Commit:
Cleanup of build.gradle based on @skalarproduktraum requested changes.

nicost commented:
I believe I addressed all the points you raised.  The CI seems to fail (no idea if that is related to these changes).  It would be awesome if you could merge (after testing), and get the CI going so that it will push new versions to the maven scijiva repo.  Thanks!

ctrueden commented:
I tested this PR on an Ubuntu Linux machine with a naive `./build.sh` and the build failed.

<details><summary>Here is the log</summary>

```
$ ./build.sh
Starting a Gradle Daemon, 1 incompatible Daemon could not be reused, use --status for details

> Configure project :
[cmake]
[make]

> Task :compileJava
Note: /home/curtis/code/clearvolume/ClearVolume/src/java/clearvolume/renderer/cleargl/ClearGLVolumeRenderer.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /home/curtis/code/clearvolume/ClearVolume/src/java/clearvolume/volume/sink/filter/gui/ChannelFilterSinkJPanel.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :javadoc
/home/curtis/code/clearvolume/ClearVolume/src/java/clearvolume/renderer/cleargl/ClearGLVolumeRenderer.java:1125: warning - @param argument "pPhase" is not a parameter name.
/home/curtis/code/clearvolume/ClearVolume/src/java/clearvolume/renderer/cleargl/ClearGLVolumeRenderer.java:1126: warning - @param argument "pClearBuffer" is not a parameter name.
/home/curtis/code/clearvolume/ClearVolume/src/java/clearvolume/renderer/ClearVolumeRendererBase.java:1238: warning - invalid usage of tag >
/home/curtis/code/clearvolume/ClearVolume/src/java/clearvolume/renderer/ClearVolumeRendererBase.java:1249: warning - @param argument "pClipBoxBounds" is not a parameter name.
/home/curtis/code/clearvolume/ClearVolume/src/java/clearvolume/renderer/ClearVolumeRendererBase.java:1271: warning - @param argument "pClipBoxBounds" is not a parameter name.
/home/curtis/code/clearvolume/ClearVolume/src/java/clearvolume/renderer/ClearVolumeRendererBase.java:2292: warning - @param argument "pByteBuffer" is not a parameter name.
/home/curtis/code/clearvolume/ClearVolume/src/java/clearvolume/renderer/ClearVolumeRendererInterface.java:267: warning - invalid usage of tag >
/home/curtis/code/clearvolume/ClearVolume/src/java/clearvolume/renderer/ClearVolumeRendererInterface.java:601: warning - invalid usage of tag >
8 warnings

> Task :cmake
Usage

  cmake [options] <path-to-source>
  cmake [options] <path-to-existing-build>
  cmake [options] -S <path-to-source> -B <path-to-build>

Specify a source directory to (re-)generate a build system for it in the
current working directory.  Specify an existing build directory to
re-generate its build system.

Run 'cmake --help' for more information.


> Task :make FAILED
make: *** No targets specified and no makefile found.  Stop.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':make'.
> Process 'command 'make'' finished with non-zero exit value 2

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 4s
8 actionable tasks: 8 executed
```

</details>

nicost commented:
Thanks Curtis!  Does that script work on the main branch?  It runs `gradle build -x test distribution`  instead of `gradle build` (which also runs a bunch of tests).  The scripts barfs for me even earlier, as I do not have cmake installed (and would like to keep it that way).  The error on the `make` task for you is also fair enough, as gradle launches `make` without arguments or a Makefile present.  My PR did not touch that stuff, and believe this was already not working before.  `gradle build` seems to succeed for you.

I don't know anything about the AppVeyor stuff and how to specify the gradle version there.  @skalarproduktraum is the plan to move from AppVeyor to github CI?

ctrueden commented:
@nicost Yeah, I was just reporting my findings for that quick test I did, in case it is somehow helpful. I'm not sure about the main branch; my first attempt to run `./build.sh` just says `./build.sh: line 3: gradle: command not found` because I don't have a system-wide gradle, and IIRC on my other machine the installed gradle was too new to successfully build it. I also tried using the `gradlew` script just now:
```
$ ./gradlew build

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine java version from '11.0.17'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
```
Using the `build` target is something I gleaned from the `.travis.yml`.

It would be helpful if the README had instructions on the intended process to build from source, since right now I am just guessing.

nicost commented:
Fully agree.  Since I don't know what the build process is or should be, I am not the one to write that README.  All I did is make it build with a gradle version that is supported on github CI (or so I believe).  @skalarproduktraum can you enlighten us?  I am happy to help, but it is a bit hard without knowing the requirements.  For now I have concentrated on building ClearVolume.version.jar, and be able to distribute it using CI, which I now believe to be possible.

skalarproduktraum committed:
Update Gradle wrapper, remove unsupported/deprecated properties from gradle.properties

skalarproduktraum committed:
Update README with build information

skalarproduktraum committed:
Appveyor CI: Remove old Gradle options

skalarproduktraum commented:
Hey @nicost and @ctrueden! Apologies again for delays, I'm severely lagging behind on my todos, so this had to wait a bit. I've made some more changes, so all scripts and Gradle stuff is now working again. Appveyor also builds. I've updated the build info on the README, too. As of now, everything looks good and builds again when running `./gradlew build`. The library then ends up in `build/libs`. [I've also made a Zulip channel for ClearVolume](https://imagesc.zulipchat.com/#narrow/stream/356698-clearvolume/topic/stream.20events), so we can discuss a bit more how this project should be handled in the future - please join! 

I'll merge this now. Thanks again @nicost for all the effort in making this compatible with Gradle 7.5.1 👍
